tictactoe.js

```diff
@@ -0,0 +1,144 @@
+
+const readlineInterface = readline.createInterface({
+  input: process.stdin,
+  output: process.stdout
+});
+
+// Ask question
+const ask = questionText => {
+  return new Promise((resolve, reject) => {
+    readlineInterface.question(questionText, resolve);
+  });
+};
+
+let values = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9];
+
+let currentPlayer = "X";
+let winner = "";
+
+const grid = gridValues => {
+  return (
+    "\n" +
+    " " +
+    gridValues[1] +
+    " | " +
+    gridValues[2] +
+    " | " +
+    gridValues[3] +
+    "\n--- --- ---\n" +
+    " " +
+    gridValues[4] +
+    " | " +
+    gridValues[5] +
+    " | " +
+    gridValues[6] +
+    "\n--- --- ---\n" +
+    " " +
+    gridValues[7] +
+    " | " +
+    gridValues[8] +
+    " | " +
+    gridValues[9] +
+    "\n\n" +
+    "Player " +
+    currentPlayer +
+    "'s turn\n" +
+    "Move to?\n"
+  );
+};
+
+const checkWinner = player => {
+  if (
+    (values[1] === player && values[2] === player && values[3] === player) ||
+    (values[4] === player && values[5] === player && values[6] === player) ||
+    (values[7] === player && values[8] === player && values[9] === player) ||
+    (values[1] === player && values[4] === player && values[7] === player) ||
+    (values[2] === player && values[5] === player && values[8] === player) ||
+    (values[3] === player && values[6] === player && values[9] === player) ||
+    (values[1] === player && values[5] === player && values[9] === player) ||
+    (values[3] === player && values[5] === player && values[7] === player)
+  ) {
+    return true;
+  }
+  return false;
+};
+
+start();
+
+async function start() {
+  let prompt = "start";
+  let spotTaken = false;
+
+  while (prompt.toLowerCase() !== "exit") {
+    if (spotTaken) {
+      prompt = await ask(
+        "\nSorry spot is already taken, try again!! \n\n" + grid(values)
+      );
+    } else {
+      prompt = await ask(grid(values));
+    }
+
+    spotTaken = false;
+    if (prompt === "1" && values[1] === 1) {
+      values[1] = currentPlayer;
+    } else if (prompt === "2" && values[2] === 2) {
+      values[2] = currentPlayer;
+    } else if (prompt === "3" && values[3] === 3) {
+      values[3] = currentPlayer;
+    } else if (prompt === "4" && values[4] === 4) {
+      values[4] = currentPlayer;
+    } else if (prompt === "5" && values[5] === 5) {
+      values[5] = currentPlayer;
+    } else if (prompt === "6" && values[6] === 6) {
+      values[6] = currentPlayer;
+    } else if (prompt === "7" && values[7] === 7) {
+      values[7] = currentPlayer;
+    } else if (prompt === "8" && values[8] === 8) {
+      values[8] = currentPlayer;
+    } else if (prompt === "9" && values[9] === 9) {
+      values[9] = currentPlayer;
+    } else {
+      spotTaken = true;
+    }
+
+    if (checkWinner("X")) {
+      winner = "X";
+      prompt = "exit";
+    } else if (checkWinner("O")) {
+      winner = "O";
+      prompt = "exit";
+    } else if (
+      values[1] !== 1 &&
```

Comment:
This works, but there are cleaner ways to write it, either in a loop or an [iteration method](http://codelikethis.com/lessons/learn-to-code-with-javascript/iteration-methods). Also, consider extracting it into its own function, named e.g. `boardFull()` or `stalemate()`, to document the "why"